### PR TITLE
[AvroToOrc] Use comma separated list for hiveRuntime properties

### PR DESCRIPTION
**Pull request by pcadabam**
Currently `hiveRuntime` properties are set in the config store as 

```
hiveRuntime : {
      mapreduce.input.fileinputformat.split.minsize=32000000
      mapreduce.input.fileinputformat.split.minsize.per.node=32000000
}

```
Since we use `TypeSafeConfig` to parse, the config  `mapreduce.input.fileinputformat.split.minsize.per.node=32000000` is overwritten by `mapreduce.input.fileinputformat.split.minsize=32000000`

Hence changing the syntax of `hiveRuntime` to a list of key value pairs.

```
hiveRuntime : [
      mapreduce.input.fileinputformat.split.minsize,32000000
      ,mapreduce.input.fileinputformat.split.minsize.per.node,32000000
]
```

@abti can you review?

### CODE REVIEW

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/dataset/ConvertibleHiveDataset.java [hunk 1]
```diff
@@ -211,16 +215,29 @@ private ConversionConfig(Config config, Table table, String destinationFormat) {
...
-      this.destinationTableProperties = ConfigUtils
-          .configToProperties(ConfigUtils.getConfig(config, DESTINATION_TABLE_PROPERTIES_KEY, ConfigFactory.empty()));
+      this.destinationTableProperties =
+          convertKeyValueListToProperties(ConfigUtils.getStringList(config, DESTINATION_TABLE_PROPERTIES_LIST_KEY));
       this.clusterBy = ConfigUtils.getStringList(config, CLUSTER_BY_KEY);
       this.numBuckets = Optional.fromNullable(ConfigUtils.getInt(config, NUM_BUCKETS_KEY, null));
-      this.hiveRuntimeProperties = ConfigUtils
-          .configToProperties(ConfigUtils.getConfig(config, HIVE_RUNTIME_PROPERTIES_KEY_PREFIX, ConfigFactory.empty()));
+
+      this.hiveRuntimeProperties =
```
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/dataset/ConvertibleHiveDataset.java]: This wont work for values that are comma separated list (for eg. search for 'comma' on this page): https://cwiki.apache.org/confluence/display/Hive/Configuration+Properties

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/publisher/HiveConvertPublisher.java [hunk 2]
```diff
@@ -119,11 +119,13 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
         directoriesToDelete.addAll(publishEntity.getCleanupDirectories());
 
         wus.setWorkingState(WorkingState.FAILED);
-        try {
-          new SlaEventSubmitter(eventSubmitter, EventConstants.CONVERSION_FAILED_EVENT, wus.getProperties()).submit();
-        } catch (Exception e) {
-          log.error("Failed while emitting SLA event, but ignoring and moving forward to curate "
-              + "all clean up comamnds", e);
+        if (!Boolean.valueOf(wus.getPropAsBoolean(PartitionLevelWatermarker.IS_WATERMARK_WORKUNIT_KEY))) {
```
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/publisher/HiveConvertPublisher.java]: This seems unrelated to hive runtime properties, what is this change for?

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/dataset/ConvertibleHiveDataset.java (hunk 1 above)
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/dataset/ConvertibleHiveDataset.java]: Fixed

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/publisher/HiveConvertPublisher.java (hunk 2 above)
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/publisher/HiveConvertPublisher.java]: This is already merged

File: gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/dataset/ConvertibleHiveDataset.java (hunk 1 above)
Comment (pcadabam) [inline gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/dataset/ConvertibleHiveDataset.java]: For comma separated values they need to be quoted. 
`"key", "value1,value2"`
